openmdao/solvers/solver.py

```diff
@@ -537,7 +598,27 @@ def solve(self):
         float
             relative error.
         """
-        return self._run_iterator()
+        fail, abs_err, rel_err = self._run_iterator()
+
+        if fail and self.options['debug_print']:
+            coord = get_formatted_iteration_coordinate()
+
+            out_str = "\n# Inputs and outputs at start of iteration '%s':\n" % coord
+            for vec_type, vec in iteritems(self._err_cache):
+                out_str += '\n'
+                out_str += '# %s %ss\n' % (vec._name, vec._typ)
+                for abs_name in sorted(vec._views.keys()):
+                    out_str += '%s = %s\n' % (abs_name, repr(vec._views[abs_name]))
+
+            print(out_str)
+
+            filename = coord.replace('._solve_nonlinear', '') + '.dat'
```

Comment:
Doesn't the iteration coord have weird symbols like pipes and colons? Not sure if those work on windows filenames.

Reply:
yes, it has non-alphanumeric characters.. does not seem to be a problem in testing.  I wanted to keep as close to possible to the iter name, but certainly willing to take a better suggestion.

Reply:
Only suggestion would be to string replace the nonalphanum chars, but if it already works in windows, then don't worry about it.

Reply:
you're right.. that would be safer and thus better. done.